spec/google-civic/client_spec.rb

```diff
@@ -37,6 +37,43 @@
       voter_info = @client.voter_info(2000, "1263 Pacific Ave. Kansas City KS", {includeOffices: true})
       voter_info.election.name.should eql "VIP Test Election"
     end
+
+    it "should retry on server error when connection_options includes retry config" do
+      retry_config = {max: 10, interval: 2}
+      stub_get("/voterinfo?electionId=2000&address=1263+Pacific+Ave.+Kansas+City+KS&key=abc123").
+       to_return({:status => 500, :body => fixture("server_error.json")},
+                 {:status => 200, :body => fixture("voter_info.json")})
+      voter_info = @client.voter_info(2000, "1263 Pacific Ave. Kansas City KS", {connection: {retry: retry_config}})
+      voter_info.election.name.should eql "VIP Test Election"
+    end
+
+    it "should retry on retriable error when connection_options includes retry config" do
+      retry_config = {max: 10, interval: 2}
+      stub_get("/voterinfo?electionId=2000&address=1263+Pacific+Ave.+Kansas+City+KS&key=abc123").
+       to_return({:status => 403, :body => fixture("retriable_error.json")},
+                 {:status => 200, :body => fixture("voter_info.json")})
+      voter_info = @client.voter_info(2000, "1263 Pacific Ave. Kansas City KS", {connection: {retry: retry_config}})
+      voter_info.election.name.should eql "VIP Test Election"
+    end
+
+    it "should not retry more than the count" do
+      retry_config = {max: 2, interval: 1}
+      stub_get("/voterinfo?electionId=2000&address=1263+Pacific+Ave.+Kansas+City+KS&key=abc123").
+       to_return({:status => 500, :body => fixture("server_error.json")},
+                 {:status => 500, :body => fixture("server_error.json")},
+                 {:status => 500, :body => fixture("server_error.json")})
+      voter_info = @client.voter_info(2000, "1263 Pacific Ave. Kansas City KS", {connection: {retry: retry_config}})
+      voter_info.error.code.should eql 500
```

Comment:
Couldn't this pass w/ a false positive since the third response is stubbed as a 500 too? Would it be better to make the third stubbed response a different status code?